htdocs/admin/content/messages.php

```diff
@@ -1,822 +1,714 @@
-			echo "<tr><th>Text</th><td><input type=\"text\" name=\"message_text\" value=\"\" size=\"20\" maxlength=\"250\" /> ";fieldqueryselbox('message_text');echo "</td></tr>";
-			echo "<tr><th style=\"width:130px;\">Flotten-ID</th><td><input type=\"text\" name=\"message_fleet_id\" value=\"\" size=\"4\" maxlength=\"250\" /></td>";
-			echo "<tr><th style=\"width:130px;\">Entitiy-ID</th><td><input type=\"text\" name=\"message_entity_id\" value=\"\" size=\"4\" maxlength=\"250\" /></td>";
-			echo "<tr><th>Gelesen</th><td><input type=\"radio\" name=\"message_read\" value=\"2\" checked=\"checked\" /> Egal
-			<input type=\"radio\" name=\"message_read\" value=\"0\" /> Nein
-			<input type=\"radio\" name=\"message_read\" value=\"1\" /> Ja</td></tr>";
-			echo "<tr><th>Rundmail</th><td><input type=\"radio\" name=\"message_massmail\" value=\"2\" checked=\"checked\" /> Egal
-			<input type=\"radio\" name=\"message_massmail\" value=\"0\" /> Nein
-			<input type=\"radio\" name=\"message_massmail\" value=\"1\" /> Ja</td></tr>";
-			echo "<tr><th>Gel&ouml;scht</th><td><input type=\"radio\" name=\"message_deleted\" value=\"2\" checked=\"checked\" /> Egal
-			<input type=\"radio\" name=\"message_deleted\" value=\"0\" /> Nein
-			<input type=\"radio\" name=\"message_deleted\" value=\"1\" /> Ja</td></tr>";
-			echo "<tr><th>Kategorie</th><td><select name=\"message_cat_id\">";
-			echo "<option value=\"\">(egal)</option>";
-			$cres = dbquery("SELECT cat_id,cat_name FROM message_cat ORDER BY cat_order;");
-			while ($carr = mysql_fetch_array($cres))
-			{
-				echo "<option value=\"".$carr['cat_id']."\">".$carr['cat_name']."</option>";
-			}
-			echo "</select></tr>";
-			echo "<tr><th>Anzahl Datens&auml;tze</th><td class=\"tbldata\"><select name=\"message_limit\">";
-			for ($x=100;$x<=2000;$x+=100)
-				echo "<option value=\"$x\">$x</option>";
-			echo "</select></td></tr>";
-
-			echo "</table>";
-			echo "<br/><input type=\"submit\" class=\"button\" name=\"user_search\" value=\"Suche starten\" /></form>";
-
-			$tblcnt = mysql_fetch_row(dbquery("SELECT count(*) FROM messages;"));
-			echo "<br/>Es sind ".nf($tblcnt[0])." Eintr&auml;ge in der Datenbank vorhanden.";
-		}
-
-	}
-?>
-
+use EtoA\Admin\AdminUser;
+use EtoA\Message\MessageCategoryRepository;
+use EtoA\Message\MessageRepository;
+use EtoA\Message\ReportRepository;
+use EtoA\User\UserRepository;
+use Symfony\Component\HttpFoundation\Request;
+
+define("USER_MESSAGE_CAT_ID",1);
+define("SYS_MESSAGE_CAT_ID",5);
+
+/** @var \EtoA\User\UserRepository $userRepository */
+$userRepository = $app['etoa.user.repository'];
+
+/** @var MessageRepository */
+$messageRepository = $app[MessageRepository::class];
+
+/** @var MessageCategoryRepository */
+$messageCategoryRepository = $app[MessageCategoryRepository::class];
+
+/** @var ReportRepository */
+$reportRepository = $app[ReportRepository::class];
+
+/** @var Request */
+$request = Request::createFromGlobals();
+
+if ($sub=="sendmsg") {
```

Comment:
I may be blind, but whee is `$sub` defined? I see a `global $sub;` in the `sendMessageForm` but this is afterwards -> I don't think it's defined here.
Also: better check with `===` instead of `==`

Reply:
See here 
https://github.com/etoa/etoa-gui/blob/88a883c338c7c5913f0e6e53a59d10c083e97bd7/htdocs/inc/bootstrap.inc.php#L50